mach-eight/reup.romulo
# Release 2.0.8 Nov 06 2024  (#267)

* Check for multiple cameras at build time

* Refactor

* Create asdw tests for move dhv camera

* Test deactivate dhv wrapper

* Fix drag manager outside of character

* Create MovePointerUtils

* Add first test to move with mouse

* create test to avoid moving dhv camera when not dragging

* Add tests to move dhv camera in touch

* Fix tests

* Remove over used prefabs and ScripHolders directory and fix tests

* Fix tests

* Improve warning message

* place camera at 50 meters from main dhv wrapper

* Restrict camera movement and fix tests

* Clean unused dependencies

* Comments addressed

* Comments addressed

* Update buildVersion to "2024-09-24" and package version to "2.0.4"

* Create TexturesManager

* Destroy a texture

* implement records

* add extra test to manager

* support protected textures

* Fix rest of the tests

* Remove event system references

* modify spaces record

* Drop class ModelInfoMessage and use JObject instead

* Refactor tests

* Zoom Dhv Camera

* Update MoveDhvCamera with serialized fields

* create first test to receive space selector list

* Send jump point's names

* return ids in space selector list

* Fix tag filters being reset on tag addition

* create first test for SpacesRecord

* move to space point

* Hide loading spinner when changing a material fails

* Create a prefab to limit player movement in external view

* Receive message in edit mediator

* Send success message

* remove dead code and assets

* Test high level feature

* Clean non-necessary

* use a Dictionary

* Refactor container

* refactor container

* fix registry

* update version date

* Comments Addressed

* fix editor tool outside the editor forlder

* DragManager Interface

* catch white space spaceJump ids

* fix tests

* Hide and show objects implementation

* Update Tests/PlayMode/TexturesManagerTest.cs



* modify edit mediator dependency injector

* resolve comments

* Comments addressed

* send not found message

* send error message if space point has no ground below

* send fail messages

* fix tests

* refactor test to be more high level

* return success messages

* fix test variables

* send interrupted message

* refactor schema

* add regex to search tag section

* deactivate filter feature

* exclusive/inclusive filters

* fix erroMessage property

* Clamp scroll input to avoid high sensitivity on trackpad

* generalize activate view mode message

* Fix appProxy tool

* Fix tests

* Add failure tests and update enum names

* Comments Addressed

* update dollhouse enum name

* escape errors in regex

* register objects in Awake

* Update package version to "2.0.5" and buildVersion to "2024-10-09"

* Improve character navigation and colliders

* Fix textures being destroyed

Textures were being destroyied when applying a texture to an object that already had that texture, and it was the only object with such texture

* add show whole building button to tag scanner tool

* delete postprocessing asset

* allow look for regex in substring filter

* deactivate house container when entering dhv

* Fix Reup prefab for dhv behaviour

* start with 40 degree angle in dhv camera

* apply antialiasing in main camera

* Fix tests

* Test movement

* remove unnused inputs

* Add zenject package to romulo

* create not passing test

* use ray utils to generate ray

* inject character position manager with zenject

* inject character position manager with zenject

* Fix test

* clean comments

* refactor instantiator

* pass first test

* failure test passing

* Improve DHV zoom tests

* apply building layer to all gameobjects inside building

* Update Editor/TagScannerTool.cs



* remove dead variables

* inject drag manager and input provider with zenject

* inject gestures manager with zenject

* Rotation working on editor

* Destroy used GameObjects in Unity tests

* test more accurate

* make some more test that are not passing

* clean

* inject character rotation keyboard with zenject

* clean comments

* Delete transformHandleObj

* make receive method a sync method

* address more warnings

* Fix texturesManager

* Sync objectIdsToTexturesRecord and texturesToObjectIdsRecord when a protected/default material is applied

* Make Error if any space jump point has no ground below

* Add gizmo

* update URP_asset

* update version

* Fix movement when gesture happens

* Fix issues

* Update build version to 2024-10-23 and package version to 2.0.7

* inputProvider Fix

* fix tests

* Fix collistion tests

* remove old file

* Fix tests in bigger screen

* Implementation of the rotation in dhv with the keyboard and mouse

* Change prefab and and scripts for rotation

* Inject building

* Inject building in rest of objects

* Fix inject in CharacterMovementKeyboard.cs

* Tests

* Remove unused space names

* Rotate Dhv with multi touch gesture

* Ensure Single Source of Truth for Object-Texture Relationships in Texture Manager

* Improve dhv rotation with multi touch gesture

* remove serialize

* Fix inputs

* Comments addressed

* Improve multi touch gesture rotation with new inputs and fix tests

* Improve tests names

* Comments addressed

* fix variable name

* make building property public

* look for building on each filter

* assign ids and tag system in editor

* move editor to be of External Installer

* inject rotation manager

* make test

* update to version 2.0.8

---------
